### ISSUE TITLE
Anchor dependency issue

### ISSUE THREAD

**Issue by glcanvas** [coral-xyz/anchor#2479]
got an dependency issue when trying to build anchor

My anchor version:

```>>> anchor --version
anchor-cli 0.24.1
```

I've initialized project:
```
anchor init solana-escrow-anchor
```

```
>>> anchor build    
BPF SDK: /Users/n.duginets/.local/share/solana/install/releases/stable-616d8d67978c325087cc5a9f33307f54eee48b6e/solana-release/bin/sdk/bpf
cargo-build-bpf child: rustup toolchain list -v
cargo-build-bpf child: cargo +bpf build --target bpfel-unknown-unknown --release
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/n.duginets/IdeaProjects/solana-escrow-anchor/programs/solana-escrow-anchor/Cargo.toml
workspace: /Users/n.duginets/IdeaProjects/solana-escrow-anchor/Cargo.toml
    Updating crates.io index
error: failed to select a version for the requirement `aho-corasick = "^1.0.0"`
candidate versions found which didn't match: 0.7.20, 0.7.19, 0.7.18, ...
location searched: crates.io index
required by package `regex v1.8.1`
    ... which satisfies dependency `regex = "^1.0"` (locked to 1.8.1) of package `anchor-attribute-access-control v0.27.0`
    ... which satisfies dependency `anchor-attribute-access-control = "^0.27.0"` (locked to 0.27.0) of package `anchor-lang v0.27.0`
    ... which satisfies dependency `anchor-lang = "^0.27.0"` (locked to 0.27.0) of package `solana-escrow-anchor v0.1.0 (/Users/n.duginets/IdeaProjects/solana-escrow-anchor/programs/solana-escrow-anchor)`
```

And got this strange error.

Looks like this is undesired behavior.

Tree's part:

```
n.duginets@macbook-C02G89JPMD6R solana-escrow-anchor % cargo tree 
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/n.duginets/IdeaProjects/solana-escrow-anchor/programs/solana-escrow-anchor/Cargo.toml
workspace: /Users/n.duginets/IdeaProjects/solana-escrow-anchor/Cargo.toml
solana-escrow-anchor v0.1.0 (/Users/n.duginets/IdeaProjects/solana-escrow-anchor/programs/solana-escrow-anchor)
└── anchor-lang v0.24.2
    ├── anchor-attribute-access-control v0.24.2 (proc-macro)
    │   ├── anchor-syn v0.24.2
    │   │   ├── anyhow v1.0.71
    │   │   ├── bs58 v0.3.1
    │   │   ├── heck v0.3.3
    │   │   │   └── unicode-segmentation v1.10.1
    │   │   ├── proc-macro2 v1.0.56
    │   │   │   └── unicode-ident v1.0.8
    │   │   ├── proc-macro2-diagnostics v0.9.1
    │   │   │   ├── proc-macro2 v1.0.56 (*)
    │   │   │   ├── quote v1.0.26
    │   │   │   │   └── proc-macro2 v1.0.56 (*)
    │   │   │   ├── syn v1.0.109
    │   │   │   │   ├── proc-macro2 v1.0.56 (*)
    │   │   │   │   ├── quote v1.0.26 (*)
    │   │   │   │   └── unicode-ident v1.0.8
    │   │   │   └── yansi v0.5.1
    │   │   │   [build-dependencies]
    │   │   │   └── version_check v0.9.4
    │   │   ├── quote v1.0.26 (*)
    │   │   ├── serde v1.0.160
    │   │   │   └── serde_derive v1.0.160 (proc-macro)
    │   │   │       ├── proc-macro2 v1.0.56 (*)
    │   │   │       ├── quote v1.0.26 (*)
    │   │   │       └── syn v2.0.15
    │   │   │           ├── proc-macro2 v1.0.56 (*)
    │   │   │           ├── quote v1.0.26 (*)
    │   │   │           └── unicode-ident v1.0.8
    │   │   ├── serde_json v1.0.96
    │   │   │   ├── itoa v1.0.6
    │   │   │   ├── ryu v1.0.13
    │   │   │   └── serde v1.0.160 (*)
    │   │   ├── sha2 v0.9.9
    │   │   │   ├── block-buffer v0.9.0
    │   │   │   │   ├── block-padding v0.2.1
    │   │   │   │   └── generic-array v0.14.7
    │   │   │   │       ├── serde v1.0.160 (*)
    │   │   │   │       └── typenum v1.16.0
    │   │   │   │       [build-dependencies]
    │   │   │   │       └── version_check v0.9.4
    │   │   │   ├── cfg-if v1.0.0
    │   │   │   ├── cpufeatures v0.2.7
    │   │   │   ├── digest v0.9.0
    │   │   │   │   └── generic-array v0.14.7 (*)
    │   │   │   └── opaque-debug v0.3.0
    │   │   ├── syn v1.0.109 (*)
    │   │   └── thiserror v1.0.40
    │   │       └── thiserror-impl v1.0.40 (proc-macro)
    │   │           ├── proc-macro2 v1.0.56 (*)
    │   │           ├── quote v1.0.26 (*)
    │   │           └── syn v2.0.15 (*)
    │   ├── anyhow v1.0.71
    │   ├── proc-macro2 v1.0.56 (*)
    │   ├── quote v1.0.26 (*)
    │   ├── regex v1.8.1
    │   │   ├── aho-corasick v1.0.1
    │   │   │   └── memchr v2.5.0
    │   │   ├── memchr v2.5.0
    │   │   └── regex-syntax v0.7.1
    │   └── syn v1.0.109 (*)
    ├── anchor-attribute-account v0.24.2 (proc-macro)
    │   ├── anchor-syn v0.24.2 (*)
    │   ├── anyhow v1.0.71
    │   ├── bs58 v0.4.0
    │   ├── proc-macro2 v1.0.56 (*)
    │   ├── quote v1.0.26 (*)
    │   ├── rustversion v1.0.12 (proc-macro)
    │   └── syn v1.0.109 (*)
```

**Comment by acheroncrypto** [comment]
Have you tried using the latest release?

**Comment by glcanvas** [comment]
yes, got the same error

**Comment by acheroncrypto** [comment]
Currently all CI tests pass https://github.com/coral-xyz/anchor/actions/runs/4850835221

Things you might want to do if you haven't already:
- Update Solana to v1.14.17
- Update Anchor to v0.27.0
- `cargo clean`
- Remove Cargo.lock

**Comment by glcanvas** [comment]
cool!
resolved with updating anchor to 0.27.0 and solana to 1.15.2
